pkg/cmd/release/create/create.go

```diff
@@ -20,54 +21,87 @@ import (
-	cmd.Flags().StringP(FlagChannel, "c", "", "Name or ID of the channel to use")
-	cmd.Flags().StringP(FlagGitRef, "r", "", "Git Reference e.g. refs/heads/main. Only relevant for config-as-code projects")
-	cmd.Flags().StringP(FlagGitCommit, "", "", "Git Commit Hash; Specify this in addition to Git Reference if you want to reference a commit other than the latest for that branch/tag.")
-	cmd.Flags().StringP(FlagPackageVersion, "", "", "Default version to use for all Packages")
-	cmd.Flags().StringP(FlagReleaseNotes, "n", "", "Release notes to attach")
-	cmd.Flags().StringP(FlagVersion, "v", "", "Version Override")
-	cmd.Flags().BoolP(FlagIgnoreExisting, "x", false, "If a release with the same version exists, do nothing rather than failing.")
-	cmd.Flags().BoolP(FlagIgnoreChannelRules, "", false, "Force creation of a release where channel rules would otherwise prevent it.")
-	cmd.Flags().BoolP(FlagPackagePrerelease, "", false, "Allow selection of prerelease packages.") // TODO does this make sense? The server is going to follow channel rules anyway isn't it?
+	flags := cmd.Flags()
+	flags.StringP(FlagProject, "p", "", "Name or ID of the project to create the release in")
+	flags.StringP(FlagChannel, "c", "", "Name or ID of the channel to use")
+	flags.StringP(FlagGitRef, "r", "", "Git Reference e.g. refs/heads/main. Only relevant for config-as-code projects")
+	flags.StringP(FlagGitCommit, "", "", "Git Commit Hash; Specify this in addition to Git Reference if you want to reference a commit other than the latest for that branch/tag.")
+	flags.StringP(FlagPackageVersion, "", "", "Default version to use for all Packages")
+	flags.StringP(FlagReleaseNotes, "n", "", "Release notes to attach")
+	flags.StringP(FlagVersion, "v", "", "Override the Release Version")
+	flags.BoolP(FlagIgnoreExisting, "x", false, "If a release with the same version exists, do nothing instead of failing.")
+	flags.BoolP(FlagIgnoreChannelRules, "", false, "Allow creation of a release where channel rules would otherwise prevent it.")
+	// cmd.Flags().BoolP(FlagPackagePrerelease, "", false, "Allow selection of prerelease packages.") // TODO does this make sense? The server is going to follow channel rules anyway isn't it?
 	// stringSlice also allows comma-separated things
-	cmd.Flags().StringSliceP(FlagPackageVersionOverride, "", []string{}, "Version Override for a specific package.\nFormat as {step}:{package}:{version}\nYou may specify this multiple times")
+	flags.StringSliceP(FlagPackageVersionSpec, "", []string{}, "Version specification a specific packages.\nFormat as {package}:{version}, {step}:{version} or {package-ref-name}:{packageOrStep}:{version}\nYou may specify this multiple times")
 
 	// we want the help text to display in the above order, rather than alphabetical
-	cmd.Flags().SortFlags = false
+	flags.SortFlags = false
+
+	// flags aliases for compat with old .NET CLI
+	util.AddFlagAliasesString(flags, FlagGitRef, FlagAliasGitRefRef, FlagAliasGitRefLegacy)
```

Comment:
@domenicsim1 this way of doing flag aliases is terrible; if you can figure out a more elegant way I'm keen for it. I tried and failed.

StackOverflow (and the cobra docs themselves somewhere) say that you can do this nicely with `setNormalizeFunc`, however I burned hours on that and it never worked :-(

https://stackoverflow.com/questions/64557741/how-to-support-an-old-flag-name-when-changing-to-a-new-flag-name-with-cobra

Viper apparently proper support, so we might get that if we pull in viper for storing config files?
https://github.com/spf13/viper#registering-and-using-aliases